### DISCUSSION TITLE
What versions of Moveit 2 are people using?

### DISCUSSION THREAD

**Discussion by riv-tnoble** [moveit/moveit2#3138] [Polls]
@sea-bass @nbbrooks 

I believe a question was briefly asked at the Maintainer Meeting yesterday regarding what versions of Moveit 2 tend to be used. I thought it might be interesting to raise a poll to gather some data as the basis for further discussion.

I've separated Rolling and Jazzy, since they're [soon to diverge](https://github.com/moveit/moveit2/issues/3116), so with these two I suppose the more relevant question would be "which of the two do you plan to use once the split occurs?". 

I've also included Iron since, despite being EOL, I can imagine there may be users who haven't yet migrated to newer versions.

**Comment by riv-tnoble** [comment]
Does Github have support for double polls? Might also be interesting to get a spread of how these version are being used (e.g. personal, production, research, etc...)

**Comment by sea-bass** [comment]
I'm not currently using MoveIt for work, but we use Humble. So that's what I will vote.

On my personal PC, I use Rolling (but I also have Jazzy as an option).

**Comment by Danilrivero** [comment]
From experience many users still even use MoveIt1 due to lack of time to properly migrate their stack to the latest version. Seeing the deprecation of ROS1 is coming soon probably many will directly go to the latest LTS. We recently migrated to Jazzy mostly due to other reasons such as GZ, but until then we have been using Humble.

**Reply by TSNoble** [reply]
> From experience many users still even use MoveIt1 due to lack of time to properly migrate their stack to the latest version. Seeing the deprecation of ROS1 is coming soon probably many will directly go to the latest LTS. We recently migrated to Jazzy mostly due to other reasons such as GZ, but until then we have been using Humble.

Yeah this is pretty accurate for us. We were using Noetic, are in the process of moving to Humble (pretty much done), and are considering upgrading to Jazzy

**Reply by sea-bass** [reply]
Most people give themselves a bit of a break between upgrades...

I will say the only really painful part of Humble -> Jazzy upgrade is that Python 3.12 now doesn't by default let you install packages into system Python. So if you need to install even a single package that is not listed as a `rosdep`, you either need to add the `--break-system-packages` flag or use virtual environments (which don't play the nicest with ROS 2 at the moment).

If you're in C++ land exclusively, no worries.

**Reply by sea-bass** [reply]
I can also make a joke about wanting to get on the Noble release of Ubuntu for the bit 😆

**Reply by sea-bass** [reply]
Specific to the MoveIt side of things, you'll find that the Humble -> Jazzy upgrade has different implementations of MoveIt Servo and Cartesian interpolator. So if you're using those, heads up.

**Reply by TSNoble** [reply]
Thanks for the info! 

We're pretty much exclusively C++, and we use mainly Pilz for on-the-fly planning, and a little OMPL to generate preplanned trajectories (I believe servo and the cartesian interpolator are separate planners?), so I think the transition shouldn't be too bad. 

I did manage to get a version of our stack launching with Jazzy, which mainly just required tweaking some of the configs (the planning adapters come to mind), and some changes to struct member names dropping underscores.

Needless to say, we likely will take a bit of a break after getting Humble stable 🙂

**Reply by sea-bass** [reply]
> (I believe servo and the cartesian interpolator are separate planners?)

They're not even planning plugins; they're just their own thing... though they do happen to use the joint group's registered IK plugins. So yes, should be no conflict there.

**Comment by okvik** [comment]
We (I) usually start with a stable release with a knowingly naive hope that this will provide some kind of a break from dealing with software. Then, bugs and/or lack of features are discovered as the project development goes on and soon enough we're on rolling or main directly. Note that this isn't specific to MoveIt2, it happens with pretty much any non-trivial code. For a small shop with practically no development resources to waste it is much easier to deal with occasional breakage of rolling than to deal with inflexibility of stable.
